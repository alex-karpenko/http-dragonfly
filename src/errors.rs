use std::io;
use thiserror::Error;

#[derive(Error)]
pub enum HttpDragonflyError {
    #[error("unable to load config file '{}': {}", .filename, .cause)]
    LoadConfigFile { filename: String, cause: io::Error },
    #[error("unable to parse config: {}", .cause.kind)]
    ParseConfigFile {
        #[from]
        cause: figment::Error,
    },
    #[error("invalid config: {}", .cause)]
    InvalidConfig { cause: String },
}

impl std::fmt::Debug for HttpDragonflyError {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", self)
    }
}
