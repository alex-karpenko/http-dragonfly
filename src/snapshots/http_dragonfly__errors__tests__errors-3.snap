---
source: src/errors.rs
expression: "HttpDragonflyError::ValidateConfig {\n    cause: \"snapshot test cause\".to_string(),\n}"
---
invalid config: snapshot test cause
